File: src/utils/storage.js
// localStorage persistence layer for DigiTox

const STORAGE_KEYS = {
  HABITS: 'digitox_habits',
  FOCUS_SESSIONS: 'digitox_focus_sessions',
  STREAK: 'digitox_streak',
  WEEKLY_DATA: 'digitox_weekly_data',
  TODAY_USAGE: 'digitox_today_usage',
  SETTINGS: 'digitox_settings',
  LAST_DATE: 'digitox_last_date',
  INTERVENTION_COUNT: 'digitox_intervention_count',
  BLOCKED_APPS: 'digitox_blocked_apps',
};

export function save(key, data) {
  try {
    localStorage.setItem(key, JSON.stringify(data));
  } catch (e) {
    console.warn('Storage save failed:', e);
  }
}

export function load(key, fallback = null) {
  try {
    const data = localStorage.getItem(key);
    return data ? JSON.parse(data) : fallback;
  } catch (e) {
    console.warn('Storage load failed:', e);
    return fallback;
  }
}

export function remove(key) {
  localStorage.removeItem(key);
}

export { STORAGE_KEYS };

File: digitox_flutter/lib/utils/storage.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';

class StorageKeys {
  static const habits = 'digitox_habits';
  static const focusSessions = 'digitox_focus_sessions';
  static const streak = 'digitox_streak';
  static const weeklyData = 'digitox_weekly_data';
  static const todayUsage = 'digitox_today_usage';
  static const settings = 'digitox_settings';
  static const lastDate = 'digitox_last_date';
  static const interventionCount = 'digitox_intervention_count';
  static const blockedApps = 'digitox_blocked_apps';
}

class Storage {
  static Future<void> save(String key, dynamic data) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      if (data is String) {
        await prefs.setString(key, data);
      } else if (data is int) {
        await prefs.setInt(key, data);
      } else if (data is bool) {
        await prefs.setBool(key, data);
      } else if (data is double) {
        await prefs.setDouble(key, data);
      } else {
        await prefs.setString(key, jsonEncode(data));
      }
    } catch (e) {
      print('Storage save failed: $e');
    }
  }

  static Future<dynamic> load(String key, {dynamic fallback}) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      if (!prefs.containsKey(key)) return fallback;

      final value = prefs.get(key);
      if (value is String) {
        try {
          return jsonDecode(value);
        } catch (_) {
          return value; // It was just a regular string
        }
      }
      return value;
    } catch (e) {
      print('Storage load failed: $e');
      return fallback;
    }
  }

  static Future<void> remove(String key) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(key);
  }
}
